demo/petstore.json

```diff
@@ -1216,7 +1216,7 @@
           "size": {
             "description": "Size scale for pets",
             "enum": ["P", "M", "G", "0"],
-            "type": "number"
```

Comment:
I totally see your point in the other test-case. But here, I would expect it to use the index, when type is `number` and no explicit values are given. Otherwise it would also be a breaking change. Or am I missing something?

Reply:
`"enum": ["P", "M", "G", "0"], type": "number"}` is not a valid json schema ...

Reply:
And, in ts, enum type present it's value.

Reply:
For a more complex example, you have a API take a enum as request body

```
export enum Size {
  P = 0,
  M = 1,
  G = 2,
  $0 = 3,
}

declare function fetch(s: Size): Promise<{}>
```

These pass type-checking:

```ts
fetch(1);
fetch(2);
fetch(3);
```

These don't

```ts
fetch('G');
fetch('P');
fetch('M');
```